tools/validator.py

```diff
@@ -104,20 +184,18 @@ def CheckFile(self, filename):
                   artifact_definition.name, filename))
           result = False
 
+        artifact_definition_supports_windows = (
+            definitions.SUPPORTED_OS_WINDOWS in (
+                artifact_definition.supported_os))
+
         for source in artifact_definition.sources:
           if source.type_indicator in (
               definitions.TYPE_INDICATOR_FILE, definitions.TYPE_INDICATOR_PATH):
-            if definitions.SUPPORTED_OS_WINDOWS in source.supported_os:
+            if (artifact_definition_supports_windows or
+                definitions.SUPPORTED_OS_WINDOWS in source.supported_os):
               for path in source.paths:
-                number_of_forward_slashes = path.count('/')
-                number_of_backslashes = path.count('\\')
-                if (number_of_forward_slashes < number_of_backslashes and
-                    source.separator != '\\'):
-                  logging.warning((
-                      'Incorrect path separator: {0:s} in path: {1:s} defined '
-                      'by artifact definition: {2:s} in file: {3:s}').format(
-                          source.separator, path, artifact_definition.name,
-                          filename))
+                if not self._CheckWindowsPath(
+                    filename, artifact_definition, source, path):
                   result = False
```

Comment:
A bit unrelated to this PR but all these could return straight away, wouldn't that be much easier to read?

Reply:
The idea is to keep checking, but return false at the end